docs: Add note regarding limitations of EMSCRIPTEN_KEEPALIVE (#18049)
emscripten-core/emscripten: site/source/docs/api_reference/emscripten.h.rst

.. c:macro:: EMSCRIPTEN_KEEPALIVE

  Tells the compiler and linker to preserve a symbol, and export it, as if you
  added it to :ref:`EXPORTED_FUNCTIONS <faq-dead-code-elimination>`.

  For example: ::

    void EMSCRIPTEN_KEEPALIVE my_function() { printf("I am being kept alive\n"); }

>   Note that this will only work if the object file in which the symbol is
>   defined is otherwise included by the linker.  This occurs when the object file
>   is part of an archive, and is not otherwise referenced. In this case any
>   symbols in the object file not be included or exported.  One way to work
>   around this limitation is to use the ``-Wl,--whole-archive`` /

Reviewer: ```suggestion
  Note that this will only work if the object file in which the symbol is
  defined is otherwise included by the linker. If the object file is part of 
  an archive and is not otherwise referenced, the linker may choose not 
  to include it and any symbols in the object file will not be included or 
  exported.  One way to work around this limitation is to use the ``-Wl,--whole-archive`` /
```

Reviewer: my 'may choose not to include it' might be better phrased as 'choose to omit it' so it's less confusing connected to the 'not included or exported' later

Author: Thanks!

I think we can say "will not include it" since this exclusions of entire objects I not optional as far I know.
